golem-cli/src/service/component.rs

```diff
@@ -64,6 +76,7 @@ pub trait ComponentService {
         uri: ComponentUri,
         project: &Option<Self::ProjectContext>,
     ) -> Result<ComponentUrn, GolemError>;
+    async fn resolve_component_name(&self, uri: &ComponentUri) -> Result<String, GolemError>;
```

Comment:
Does `resolving` component_name based on `component_uri` depend on the instance of `ComponentService`? Should this be really abstract?